geomstats/geometry/invariant_metric.py

```diff
@@ -124,18 +124,10 @@ def inner_product(self, tangent_vec_a, tangent_vec_b, base_point=None):
                 tangent_vec_b,
                 base_point)
 
-        jacobian = self.group.jacobian_translation(base_point)
-        inv_jacobian = self.group.inverse(jacobian)
-        if self.left_or_right == 'left':
-            tangent_vec_a_at_id = self.group.compose(
-                inv_jacobian, tangent_vec_a)
-            tangent_vec_b_at_id = self.group.compose(
-                inv_jacobian, tangent_vec_b)
-        elif self.left_or_right == 'right':
-            tangent_vec_a_at_id = self.group.compose(
-                tangent_vec_a, inv_jacobian)
-            tangent_vec_b_at_id = self.group.compose(
-                tangent_vec_b, inv_jacobian)
+        tangent_translation = self.group.tangent_translation_map(
+            base_point, left_or_right=self.left_or_right, inverse=True)
+        tangent_vec_a_at_id = tangent_translation(tangent_vec_a)
```

Comment:
I wonder if we could use the parallel transport naming for the group left-invariant connection? Something like `left_parallel_transport`?

Reply:
I would argue not to do that, because this is only true with a certain connection, not necessarily the canonical one: on SOn, with the bi-invariant connection, the parallel transport is dL_h dR_h where h = sqrt(g). Plus parallel transport needs a connection or a metric to be defined, where as the left translation only needs the smooth group structure.

Reply:
Ok!